src/lazy/expanded/macro_evaluator.rs

```diff
@@ -1164,36 +1200,23 @@ impl<'top> TemplateExpansion<'top> {
         }
     }
 
+    pub fn definition(&self) -> TemplateMacroRef<'top> {
+        self.template
+    }
+
     pub(crate) fn next<'data: 'top, D: Decoder>(
         &mut self,
         context: EncodingContextRef<'top>,
         environment: Environment<'top, D>,
     ) -> IonResult<MacroExpansionStep<'top, D>> {
         let expressions = self.template.body().expressions();
-        let value_expr = match expressions.get(self.step_index) {
+        let value_tdl_expr = match expressions.get(self.step_index) {
             None => return Ok(MacroExpansionStep::FinalStep(None)),
             Some(expr) => expr,
         };
+        let value_expr = value_tdl_expr.to_value_expr(context, environment, self.template);
```

Comment:
🗺️ I factored out the `match` logic needed to transform a `TemplateBodyExpr` to a `ValueExpr`, which simplified expansion code in a number of places including this one.